tools/install-powershell.ps1

```diff
@@ -85,25 +107,75 @@ try {
         $metadata = Invoke-RestMethod https://api.github.com/repos/powershell/powershell/releases/latest
         $release = $metadata.tag_name -replace '^v'
 
-        $packageName = "PowerShell-${release}-win-${architecture}.zip"
+        $packageName = if ($IsWinEnv) {
+            "PowerShell-${release}-win-${architecture}.zip"
+        } elseif ($IsLinuxEnv) {
+            "powershell-${release}-linux-${architecture}.tar.gz"
+        } elseif ($IsMacOSEnv) {
+            "powershell-${release}-osx-${architecture}.tar.gz"
+        }
+
         $downloadURL = "https://github.com/PowerShell/PowerShell/releases/download/v${release}/${packageName}"
         Write-Verbose "About to download package from '$downloadURL'" -Verbose
 
         $packagePath = Join-Path -Path $tempDir -ChildPath $packageName
         Invoke-WebRequest -Uri $downloadURL -OutFile $packagePath
 
-        Expand-Archive -Path $packagePath -DestinationPath $Destination
+        if ($IsWinEnv) {
+            Expand-Archive -Path $packagePath -DestinationPath $Destination
```

Comment:
Should we delete the compressed package after expansion?

Reply:
The compressed package will be deleted. The compressed package is downloaded to `$tempDir` which will be deleted in the finally block at the end of the script.